src/app/content/highlights/components/EditCard.tsx

```diff
@@ -68,6 +72,13 @@ const EditCard = React.forwardRef<HTMLElement, Props>((
       onBlur();
     }
   };
+ 
+  React.useEffect(() => {
+    if (!isFocused && hasUnsavedHighlight) {
```

Comment:
why do you need `hasUnsavedHighlight` here? can't you just reset everything whenever you lose focus?

also you can probably call `cancelEditing` instead of duplicating this